src/evm-block-extractor/src/block_extractor.rs

```diff
@@ -57,7 +57,15 @@ impl BlockExtractor {
                 Duration::from_secs(request_time_out_secs),
                 client.get_full_blocks_by_number(block_numbers, batch_size),
             )
-            .await??;
+            .await?;
+
+            let evm_blocks = match evm_blocks {
```

Comment:
I don't agree with this. A missing block is a error situation that must stop the processing

Reply:
What if the block extractor was started when we removed some blocks, that will ever err out; how are we supposed to resolve it ?

Reply:
Maybe it could retrieve the block number at this point to check who is out of sync

Reply:
The block extractor must have all blocks, this is why we have it. If we miss some blocks we are in big troubles because we cannot recreate the blockchain from scratch

Reply:
OK